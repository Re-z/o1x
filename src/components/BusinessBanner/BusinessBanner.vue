<template>
    <div class="flex items-center justify-center bg-green-900 px-2 py-4">
        <img
            width="100"
            height="100"
            src="https://place-hold.it/100x100"
            alt="placheholder text"
        />
        <div class="mx-5">
            <p class="text-2xl text-white">{{ $t('businessBannerTitle') }}</p>
            <p class="text-l text-white">{{ $t('businessBannerText') }}</p>
        </div>
        <Button :text="$t('details')" />
    </div>
</template>

<script setup lang="ts">
    // imports
    import Button from '../generic/Button/Button.vue'
</script>
