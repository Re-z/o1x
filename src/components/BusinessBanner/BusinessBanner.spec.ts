import { describe, test, expect } from 'vitest'
import { mountComponent } from '@/__tests__/setup'
import BusinessBanner from '@/components/BusinessBanner/BusinessBanner.vue'

describe('BusinessBanner.vue tests', () => {
    test('renders correctly (snapshot)', () => {
        const wrapper = mountComponent(BusinessBanner)
        expect(wrapper.html()).toMatchSnapshot()
    })
})
