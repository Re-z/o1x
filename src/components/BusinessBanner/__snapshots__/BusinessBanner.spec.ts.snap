// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BusinessBanner.vue tests > renders correctly (snapshot) 1`] = `
"<div class="flex items-center justify-center bg-green-900 px-2 py-4"><img width="100" height="100" src="https://place-hold.it/100x100" alt="placheholder text">
  <div class="mx-5">
    <p class="text-2xl text-white">Маєш бізнес?</p>
    <p class="text-l text-white">Переглянь актуальні пропозиції для бізнесу</p>
  </div><!-- TODO: optimize -->
  <div class="button-wrapper">
    <div class="transform cursor-pointer rounded-lg bg-yellow-500 px-4 py-2 text-lg transition duration-500 ease-in-out hover:bg-yellow-600 active:scale-95" style="color: rgb(255, 255, 255);">Деталі</div>
    <!--v-if-->
  </div>
</div>"
`;
